zetaclient/zeta_supply_checker.go

```diff
@@ -0,0 +1,236 @@
+package zetaclient
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	sdkmath "cosmossdk.io/math"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/rs/zerolog"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	"github.com/zeta-chain/zetacore/zetaclient/config"
+)
+
+type ZetaSupplyChecker struct {
```

Comment:
Can we define an interface for `ZetaSupplyChecker`?
We're going to use interfaces for ZetaClient unit tests which will be one of the next milestone for the ZetaClient